### PR TITLE
cele_threads: simple thread pool API interface of threads_pool

### PR DESCRIPTION
**Pull request by lishulong**
# Description
公共thread pool 封装。
提供简单接口，提供trait以及错误类型。
方便换库。

Fixes # (issue)

## Type of change
New feature

# Checklist:

- [x] **Style**:       My code follows the **style guidelines** of this project
- [x] **Self-review**: I have performed a **self-review** of my own code
- [x] **Comment**:     I have **commented my code**, particularly in hard-to-understand areas
- [x] **Doc**:         I have made corresponding changes to the **documentation**
- [x] **No-warnings**: My changes generate **no new warnings**
- [ ] **Add-test**:    I have added **tests** that prove my fix is effective or that my feature works
- [x] **Pass**:        New and existing **unit tests pass** locally with my changes
- [x] **Dep**:         Any **dependent** changes have been merged and published in downstream modules

### CODE REVIEW

File: components/cele_threads/src/cele_threads.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
+extern crate quick_error;
+extern crate threads_pool;
+
+use quick_error::quick_error;
+use std::convert::From;
+use std::fmt;
+use threads_pool::{ExecutionError, PoolManager, ThreadPool};
+
+quick_error! {
+    pub enum CTError {
```
Comment (drmingdrmer) [inline components/cele_threads/src/cele_threads.rs]: 类型还是用全称吧

File: components/cele_threads/src/cele_threads.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            CTError::Timeout(e) => execute_error_to_str(e),
+            CTError::InternalErr(e) => execute_error_to_str(e),
+        };
+
+        write!(f, "CTError::{}", msg)
+    }
+}
+
+impl From<ExecutionError> for CTError {
+    fn from(e: ExecutionError) -> Self {
```
Comment (wenbobuaa) [inline components/cele_threads/src/cele_threads.rs]: 可以直接在定义的地方实现 form : 
```
pub enum CTError {
        Timeout(err: ExecutionError) {
            from()
        }
        InternalErr (err: ExecutionError) {
            from()
        }
}
```

忘了有没有实现 fmt 的 quick 方式了

File: components/cele_threads/src/cele_threads.rs (hunk 1 above)
Comment (lishulong) [reply components/cele_threads/src/cele_threads.rs]: ok

File: components/cele_threads/src/cele_threads.rs (hunk 2 above)
Comment (lishulong) [reply components/cele_threads/src/cele_threads.rs]: 不支持这样写，我想要的是从ExecutionError的不同的value来生成不同的CTError的value。
```
quick_error! {
    #[derive(Debug)]
    enum MyError {
    ¦   Err1(val: String) {
    ¦   ¦   display("hello1 {}", val)
    ¦   ¦   // from String to MyError
    ¦   ¦   from()
    ¦   }
    ¦   Err2(val: String) {
    ¦   ¦   display("hello2 {}", val)
    ¦   ¦   from()
    ¦   }
    }
}
这样会报错
```

Comment (wenbobuaa) [reply components/cele_threads/src/cele_threads.rs]: 嗷，那就不能用.. 还是保持原来吧

Comment (lishulong) [reply components/cele_threads/src/cele_threads.rs]: -> 忘了有没有实现 fmt 的 quick 方式了

应该没有。我自己加了一个fmt，提示了一个error
```
no rules expected the token `ERROR_CHECK`
```

**Pull request merged**